test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1

```diff
@@ -212,3 +212,94 @@ Describe "Members of System.Type" -Tags "CI" {
         [MyType].ImplementedInterfaces | Should -Be System.Collections.IEnumerable
     }
 }
+
+Describe "Hash expression with if statement as value" -Tags "CI" {
+    BeforeAll {
+        # With no extra new lines after if-statement
+        $hash1 = @{
+            a = if (1) {'a'}
+            b = 'b'
+            c = if (0) {2} elseif (1) {'c'}
+            d = 'd'
+            e = if (0) {2} elseif (0) {2} else {'e'}
+            f = 'f'
+            g = if (0) {2} else {'g'}
+            h = 'h'
+        }
+
+        # With extra new lines after if-statement
```

Comment:
Perhaps the comment should be more stringent
`Caution!!! Don't remove extra new lines after if-statement!`